includes/Traits/Assets.php

```diff
@@ -58,22 +65,36 @@ protected function get_asset_metadata( $handle ) {
 	 * @param string $script_handle Handle of script.
 	 * @param array  $script_dependencies Array of extra dependencies.
 	 *
-	 * @return void
+	 * @return array Asset metadata of the script.
 	 */
 	public function register_script( $script_handle, array $script_dependencies = [] ) {
-		$asset        = $this->get_asset_metadata( $script_handle );
-		$dependencies = array_merge( $asset['dependencies'], $script_dependencies );
-		$version      = $asset['version'];
+		$base_script_path = WEBSTORIES_PLUGIN_DIR_URL . 'assets/js/';
+
+		$asset   = $this->get_asset_metadata( $script_handle );
+		$version = $asset['version'];
+
+		// Register any chunks of $script_handle first.
+		foreach ( $asset['js'] as $script_chunk ) {
+			wp_register_script(
+				$script_chunk,
+				$base_script_path . $script_chunk . '.js',
+				[],
+				$version
```

Comment:
Is `$in_footer` omitted by design here?

Reply:
Just forgotten I think :-)

Should be `true` for sure.

Reply:
Good catch!

Reply:
Why `$in_footer = true` if the main script sets `false`? Shouldn't all scripts have the same loading semantics?

Reply:
Didn't realize the main one had `false`. We should change that (and ideally make customizable in the future)